InstGen/ModelPrinter.cpp

```diff
@@ -326,13 +329,16 @@ void ModelPrinter::analyzeEqualityAndPopulateDomain()
 
     ALWAYS(ecElIt.hasNext());
     unsigned firstFunc = ecElIt.next();
+    //TODO is the below correct? AYB
```

Comment:
Any update on this? Or should the todo carry over to after merge?

Reply:
There is also one additional "TODO" slightly above.

Reply:
The confusion comes from an ignorance of what exactly `IntUnionFind`. Previously it took the set of all function symbols and did stuff (which I don't understand). It now takes a set which includes functional symbols and type constructors. Is it sufficiently to filter out the type constructors on line 332, or could there be other issues?

Reply:
This is not helping me much to resolve this.

How about we try running Instgen on some satisfiable problems with proof printing on to see whether it does something reasonable?